sycl/include/CL/sycl/half_type.hpp

```diff
@@ -400,6 +400,37 @@ class half {
     half r = *this;
     return -r;
   }
+
+// Operator +, -, *, /
+#define OP(op, op1)                                                            \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const half &lhs,               \
+                                                const half &rhs) {             \
+    half rtn = lhs;                                                            \
+    rtn op1 rhs;                                                               \
+    return rtn;                                                                \
+  }                                                                            \
+  template <typename T,                                                        \
+            typename = std::enable_if_t<std::is_arithmetic<T>::value>>         \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const half &lhs,               \
+                                                const T &rhs) {                \
+    half rtn = lhs;                                                            \
+    rtn op1 rhs;                                                               \
+    return rtn;                                                                \
+  }                                                                            \
+  template <typename T,                                                        \
+            typename = std::enable_if_t<std::is_arithmetic<T>::value>>         \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const T &lhs,                  \
```

Comment:
Let's say 'T' is 'double'. "double double_val = 10.0;"
Shouldn't the operation "auto res = double_val + half_val;" return double?

Reply:
I have made updated it so it returns a double type in this situation